UpdateDependencies.ps1

```diff
@@ -0,0 +1,125 @@
+﻿#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+function EnsureEnvironmentVariable([string]$envVarName)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        throw "Can't find environment variable '$envVarName'"
+    }
+}
+
+function SetEnvIfDefault([string]$envVarName, [string]$value)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        [Environment]::SetEnvironmentVariable($envVarName, $value)
+    }
+}
+
+EnsureEnvironmentVariable "GITHUB_USER"
+EnsureEnvironmentVariable "GITHUB_EMAIL"
+EnsureEnvironmentVariable "GITHUB_PASSWORD"
+
+SetEnvIfDefault 'COREFX_VERSION_URL' 'https://raw.githubusercontent.com/dotnet/versions/master/dotnet/corefx/master/Latest.txt'
+SetEnvIfDefault 'GITHUB_ORIGIN_OWNER' $env:GITHUB_USER
+SetEnvIfDefault 'GITHUB_UPSTREAM_OWNER' 'dotnet'
+SetEnvIfDefault 'GITHUB_PROJECT' 'corefx'
+SetEnvIfDefault 'GITHUB_UPSTREAM_BRANCH' 'master'
+
+
+$CoreFxLatestVersion = Invoke-WebRequest $env:COREFX_VERSION_URL -UseBasicParsing
+$CoreFxLatestVersion = $CoreFxLatestVersion.ToString().Trim()
+
+# Updates the dir.props file with the latest CoreFX build number
+function UpdateValidDependencyVersionsFile
+{
+    if (!$CoreFxLatestVersion)
+    {
+        Write-Error "Unable to find latest CoreFX version at $env:COREFX_VERSION_URL"
+        return $false
+    }
+
+    $DirPropsPath = "$PSScriptRoot\dir.props"
```

Comment:
I don't have a particular issue with what you have here but I thought I would at least share how to do the same thing in powershell cmdlets as well (I wrote this a while ago as when I was looking at trying to automate this).

```
$dirprops = Get-Content dir.props | % { $_ -replace "<CoreFxExpectedPrerelease>(.*)</CoreFxExpectedPrerelease>","<CoreFxExpectedPrerelease>$buildNumber</CoreFxExpectedPrerelease>" }
Set-Content dir.props $dirprops
```